packages/compass-import-export/src/import/import-utils.ts

```diff
@@ -125,3 +128,27 @@ export function processParseError({
     }
   }
 }
+
+export function makeDocStatsStream() {
+  let biggestDocSize = 0;
+  const stream = new Transform({
+    objectMode: true,
+    transform: function (doc, encoding, callback) {
+      try {
+        const docString = JSON.stringify(doc);
```

Comment:
Hmmm, I was expecting that we would be able to get the stringified version from importers so we can avoid doing the double work. Both import-json and import-csv has access to the chunk that is parsed into the doc, right?

Reply:
Maybe a progress callback is a better place for this? We already use to to report bytes and docs processed and it is triggered for every document processed

Reply:
yea, right before these chunks are passed into parser streams but what I noticed is that those chunks could be a group of multiple docs. It is the parser streams that extracts the meaningful docs out of them.